pkg/asset/store/store.go

```diff
@@ -125,6 +142,44 @@ func (s *storeImpl) DestroyState() error {
 	return nil
 }
 
+// validateAssets checks for expired certificates and removes the expired certificates from
+// the installer state file.
+func (s *storeImpl) validateAssets(assets map[string]json.RawMessage) error {
+	for assetName, data := range assets {
+		storedAssetInfo := storedAsset{}
+		err := json.Unmarshal(data, &storedAssetInfo)
+		if err != nil {
+			logrus.Warnf("failed to unmarshal asset %s from state file: %s", assetName, err.Error())
+			continue
+		}
+
+		for _, file := range storedAssetInfo.FileList {
+			if filepath.Ext(file.Filename) == ".crt" && file.Data != nil {
+				data := file.Data
+				for {
+					block, rest := pem.Decode(data)
+					if block == nil {
+						break
+					}
+
+					cert, err := x509.ParseCertificate(block.Bytes)
+					if err == nil {
+						if time.Now().UTC().After(cert.NotAfter) {
+							return fmt.Errorf("state file certificate %s expired at %s", path.Base(file.Filename), cert.NotAfter.Format(time.RFC3339))
```

Comment:
What action could the user take to fix this if it occurs?

Reply:
As of right now they would have to run `openshift-install destroy cluster` so that the assets are cleaned up. This would leave only the log file but no state file.

Reply:
@patrickdillon what do you think?